src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -825,9 +878,7 @@ void ShenandoahHeap::notify_heap_changed() {
   // Update monitoring counters when we took a new region. This amortizes the
   // update costs on slow path.
   monitoring_support()->notify_heap_changed();
-
-  // This is called from allocation path, and thus should be fast.
-  _heap_changed.try_set();
+  _heap_changed.set();
```

Comment:
Why not leave `try_set` intact? This alleviates contention on the shared counter, AFAICS.

Reply:
https://bugs.openjdk.org/browse/JDK-8344797